File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    closet = {}
    result = 1
    
    for elements in clothes:
        key = elements[1]
        value = elements[0]
        if key in closet:
            closet[key].append(value)
        else:
            closet[key] = [value]
            
    for key in closet.keys():
        result = result * (len(closet[key])+1)
    
    return result - 1

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> m;
    
    for (auto cv: clothes) {
        m[cv[1]]++;
    }
    
    int answer = 1;
    for (auto it: m) {
        answer *= (it.second + 1);
    }
    
    return answer - 1;
}